packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+		const summarizer = await Summarizer.create(loader, absoluteUrl);
+		await waitForSummarizerConnection(summarizer);
+		return summarizer;
+	}
+
+	async function waitForSummarizerConnection(summarizer: ISummarizer): Promise<void> {
+		const runtime = (summarizer as any).runtime as ContainerRuntime;
+		if (!runtime.connected) {
+			return new Promise((resolve) => runtime.once("connected", () => resolve()));
+		}
+	}
+
+	it("Can load a paused container", async () => {
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+			},
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+		const initialValue = dataObject2.value;
+
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		dataObject1.increment();
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+		assert.strictEqual(
+			initialValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the initial value",
+		);
+	});
+
+	it("Can load a paused container at a specific sequence number", async () => {
+		// Min 5 ops, max 15 ops
+		const numIncrement = Math.round(Math.random() * 10) + 5;
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		// Record sequence number we want to pause at, and the expected value at that sequence number
+		const sequenceNumber = container1.deltaManager.lastSequenceNumber;
+		const expectedValue = dataObject1.value;
+
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+				opsBeforeReturn: "sequenceNumber",
+			},
+			[LoaderHeader.sequenceNumber]: sequenceNumber,
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should be at the specified sequence number",
+		);
+		assert.strictEqual(
+			expectedValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the expected value",
+		);
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
```

Comment:
A stronger assertion here and throughout would be to assert for each container the specific sequence number you expect it to be, rather than just that they are not equal to one another.  Both in terms of being a higher bar to satisfy but also explicitly/directly expressing the specific guarantee the API provides.